input/input-cervantes.h

```diff
@@ -0,0 +1,66 @@
+/*
+    KOReader: Bq Cervantes input abstraction for Lua
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef _KO_INPUT_CERVANTES_H
+#define _KO_INPUT_CERVANTES_H
+
+#define USBHOST_DEVPATH "/devices/platform/usb_host"
+#include "libue.h"
+
+static void sendEvent(int fd, struct input_event* ev) {
+    if (write(fd, ev, sizeof(struct input_event)) == -1) {
+        fprintf(stderr, "[ko-input]: Failed to generate fake event.\n");
+    }
+}
+
+static void generateFakeEvent(int pipefd[2]) {
+    int re;
+    struct uevent_listener listener;
+    struct uevent uev;
+    struct input_event ev;
+
+    close(pipefd[0]);
+
+    ev.type = EV_KEY;
+    ev.value = 1;
+
+    re = ue_init_listener(&listener);
+    if (re < 0) {
+        fprintf(stderr, "[ko-input]: Failed to initilize libue listener, err: %d\n", re);
```

Comment:
initialize ;)

NOTE: It's not actually your typo, so if you can fix the same one in input-kobo while you're there, that'd be great ;).

Reply:
do you mean re > 0 ?

Reply:
No, I meant `initilize` -> `initialize` in the fprintf ;).

Reply:
ah, that makes sense :+1: